pkg/obfuscator/generator.go

```diff
@@ -0,0 +1,33 @@
+package obfuscator
+
+import (
+	"fmt"
+
+	"k8s.io/klog/v2"
+)
+
+// generator consists of the required fields for the consistent,static obfuscations and the count of the obfuscations
+// This implements the methods static, consistent inorder to return the required replacement based on the replacementType
+type generator struct {
+	template string
+	static   string
+	count    int
+}
+
+func (g *generator) generateConsistentReplacement(_ string) string {
+	g.count++
+	if g.count > maximumSupportedObfuscations {
+		klog.Exitf("maximum number of obfuscations exceeded: %d", maximumSupportedObfuscations)
+	}
+	r := fmt.Sprintf(g.template, g.count)
+	return r
+}
+
+func (g *generator) generateStaticReplacement(_ string) string {
+	return g.static
+}
+
+// NewGenerator creates a generator objects and populates with the provided arguments
+func NewGenerator(template, static string, count int) *generator {
```

Comment:
An exported function should not return an unexported type.

Reply:
Since `count` will always start from 0 other than the tests it should not be passed as a parameter.

Reply:
You don't need to initialize it either because that's the [default value](https://tour.golang.org/basics/12).

Reply:
Yeah, I will address these